src/asl.ts

```diff
@@ -2414,28 +2627,48 @@ export class ASL {
     };
   }
 
-  public conditionState(cond: Condition): ASLGraph.OutputSubState {
-    const tempHeap = this.newHeapVariable();
+  /**
+   * When the condition is true, executes the true states and assigns the left output to a temp variable.
+   * When the condition is false, executes the true states and assigns the right out to a temp variable.
+   *
+   * @param trueState - states to execute when the condition is true. Default: returns true
+   * @param falseState - states to execute when the condition is false. Default: returns false
+   *
+   * @returns a {@link ASLGraph.OutputSubState} that executes true when the value is true and false when the value is false.
+   */
+  public conditionState(
+    node: FunctionlessNode | undefined,
+    cond: Condition,
+    trueState?: ASLGraph.NodeResults,
+    falseState?: ASLGraph.NodeResults,
```

Comment:
Why would these ever be `undefined`?

Reply:
Defaults to return true and false constants, which was the original contract of this function.